src/main/java/org/mondemand/Client.java

```diff
@@ -618,9 +658,14 @@ public void increment (StatType type, String key, int value) {
    * @param context context
    * @param keyType key
    * @param value value
+   * @throws MondemandException
    */
-  public void increment(ContextList context, String keyType, long value)
+  public void increment(ContextList context, String keyType, long value) throws MondemandException
   {
+    if(!isKeyValid(keyType)) {
```

Comment:
I don't think we should do this check here.  It's going to be a lot of additional work that this will be doing when instead we can check the keys just before we send them to get the same effect.  True, we won't catch the bad increments when they happen, but I don't think it's worth the overhead this will add

Reply:
I spoke to Nick about this. Se we can verify the keys in two different places:
- do it during increment.
- don't bother checking the key during increment, only do it right before emission. 

The problem with the first one is what Nick mentioned in his comment, the problem with the second one is if we throw exception the we will be killing the thread that does the emission once a minute, which is worse that the first problem (and logging stuff as opposed to throwing exception is not something we want to do). 

So we decided to define a set to keep the keys that have already been verified and in the `isKeyValid()` we first check that set. This way we won't be initiating and calling the `matcher` for every `increment()` call. I will change the code accordingly.